src/Interpreters/RowRefs.h

```diff
@@ -112,6 +113,17 @@ struct RowRefList : RowRef
             }
         }
 
+        Batch * getBatch()
+        {
+            return batch;
+        }
+
+        void nextBatch()
+        {
+            batch = batch->next;
```

Comment:
This invalidates the existing `position, using `operator ++` and `nextBatch()` together is not viable.